inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/transform/StreamTransformServiceImpl.java

```diff
@@ -139,16 +142,35 @@ public Integer save(TransformRequest request, UserInfo opInfo) {
     }
 
     @Override
-    public List<TransformResponse> listTransform(String groupId, String streamId) {
+    public PageResult<TransformResponse> listTransform(TransformPageRequest request) {
+        String groupId = request.getInlongGroupId();
+        String streamId = request.getInlongStreamId();
         LOGGER.debug("begin to fetch transform info by groupId={} and streamId={} ", groupId, streamId);
-        Preconditions.expectNotBlank(groupId, ErrorCodeEnum.GROUP_ID_IS_EMPTY);
-        List<StreamTransformEntity> entityList = transformMapper.selectByRelatedId(groupId, streamId, null);
-        if (CollectionUtils.isEmpty(entityList)) {
-            return Collections.emptyList();
-        }
+        PageHelper.startPage(request.getPageNum(), request.getPageSize());
+        PageResult<TransformResponse> pageResponse = new PageResult<>();
+        pageResponse.setList(getTransform(groupId, streamId));
```

Comment:
For the scalability, it's better to make a **selectByCondition** method in **StreamTransformMapper** which accept **TransformPageRequest** and select entities by those condition defined in page request.

Reply:
Done.